src/Disciplina/Launcher/Mode.hs

```diff
@@ -1,49 +1,58 @@
-{-# LANGUAGE TemplateHaskell #-}
+{-# LANGUAGE TemplateHaskell        #-}
+{-# LANGUAGE TypeFamilyDependencies #-}
 
--- | Module contains the definition of WorkMode and its implementations.
+-- | Module contains the definition of basic WorkMode and its implementations.
 
 module Disciplina.Launcher.Mode
        (
          -- * Constraints
          BasicWorkMode
 
          -- * Implementations
-       , BasicRealMode
-       , BasicRealContext (..)
+       , RealMode
+       , NodeContext (..)
+       , CustomContext
+       , ncCustomCtx
        ) where
 
 import Universum
 
 import Control.Lens (makeLenses)
-import Mockable (MonadMockable, Production (..))
+import Control.Monad.Trans.Control (MonadBaseControl)
+import Disciplina.Async (Forall, Pure)
 import System.Wlog (HasLoggerName (..), LoggerName, WithLogger)
 
-import Disciplina.DB.Class (MonadDB)
-
 ---------------------------------------------------------------------
 -- WorkMode classes
 ---------------------------------------------------------------------
 
 -- | Set of typeclasses which define basic capabilities of Disciplina node
 type BasicWorkMode m =
     ( WithLogger m
-    , MonadMockable m
     , MonadIO m
-    , MonadDB m
+    , MonadBaseControl IO m
+    , Forall (Pure m)  -- allows to use lifted-async
     )
 
 ---------------------------------------------------------------------
 -- WorkMode implementations
 ---------------------------------------------------------------------
 
-data BasicRealContext = BasicRealContext
-    { _brcLoggerName :: LoggerName
+-- | For node role returns related context.
+type family CustomContext r = cc | cc -> r
```

Comment:
This is an absolutely useless abstraction. You could just define `EducatorCustomContext = EducatorCustomContext { ecContext :: NodeContext, whatever :: WhateverContext }` instead, and I can't see any reason not to do that.

Reply:
Then one will need to define basic typeclasses like `MonadLogging` for both `EducatorMode` and `WitnessMode` in the same way but separately.

Reply:
@Martoon-00 
We can still do this and use approach with `Has*` lenses to not define such typeclasses. I personally do not like this approach too much, but I admit that it's more general than current one. We'll discuss it with @volhovm in more detail tomorrow.